src/System.Security.Cryptography.Pkcs/src/System/Security/Cryptography/Pkcs/SignerInfo.cs

```diff
@@ -108,6 +108,83 @@ public SignerInfoCollection CounterSignerInfos
 
         public Oid SignatureAlgorithm => new Oid(_signatureAlgorithm);
 
+        public void AddUnsignedAttribute(AsnEncodedData unsignedAttribute)
```

Comment:
Our CounterSigner does add a new attribute set per counter-signer, to maintain parity with what NetFx was already doing.

Please try to find some other frameworks and get an idea of what they would do here.

It seems weird that if someone prepares them ahead of time we will build

```
UnsignedAttributes
    AttributeSet
        oid
        values
            value0
            value1
```

But that if they do it individually with AddUnsignedAttribute it becomes

```

UnsignedAttributes
    AttributeSet
        oid
        values
            value0
    AttributeSet
        oid
        values
            value1
```

Reply:
Ok, I only did that to match ComputeCounterSignature, will check what other frameworks do

Reply:
@bartonjs other options is we could add `bool mergeWithExisting` (or something like that) to let user choose

Reply:
I'm good with a default-value parameter for it.  Though we still want to do some peer research to find out if there's a clear-cut winner for which way the default should go.

Reply:
@bartonjs Bouncy Castle allows setting Asn1Set directly in AttributeTable which allows both. In OpenSSL you also provide the whole attribute. I think only counter signatures are expected to be duplicate. We got 3 options:
- leave as is, pick whichever convention and if we got request with a valid scenario we add bool overload
- add a bool overload (argument might be confusing) - we might want to rename API to something like AddAttributeValue and call bool something like `mergeWithExistingAttribute` to make it less confusing
- come back to my previous design which take CryptographicAttributeObject

I think first and second option sound reasonably good - I'm not convinced bool will be super useful but we can add it just in case. Thoughts?